### PR TITLE
docs: Update template and field path syntax

### PR DESCRIPTION
**Pull request by spencergilbert**
Field path notation has been replaced by the shared lookup code. Problematically we don't report errors from these very well see https://github.com/vectordotdev/vector/issues/13033 for the solution there.

Technically these don't require (today) a leading `.` in the path expression, but that may change in the future and, to me, it's less confusing to have two different ways of handling lookups.

### CODE REVIEW

**Review by jszwedko — COMMENTED** [review]
Changes look good to me. Seems like deploy is failing, but I can review again when the deploy is up.

**Review by jszwedko — APPROVED** [review]
Looks good!

File: website/content/en/docs/reference/configuration/template-syntax.md [hunk 1]
```diff
@@ -102,23 +105,12 @@ Vector doesn't currently support fallback values. [Issue 1692][1692] is open to
...
-
-### Nested fields
-
-Nested values can be accessed using Vector's [field notation syntax][paths]:
-
-```toml
-option = "{{ parent.child[0] }}"
-```
-
+metric is incremented with a `error_type` tag of `template_failed`.
```
Comment (neuronull) [inline website/content/en/docs/reference/configuration/template-syntax.md:108]:
```suggestion
metric is incremented with an `error_type` tag of `template_failed`.
```

**Review by neuronull — APPROVED** [review]
Mostly reviewed for spelling/grammar.

File: website/content/en/docs/reference/configuration/template-syntax.md (hunk 1 above)
Comment (spencergilbert) [reply website/content/en/docs/reference/configuration/template-syntax.md]: Bah, originally I had written it as `incremented with a tag...` - thanks for the catch.

**Review by fuchsnj — APPROVED** [review]

**Pull request merged**